src/drive/actions/async.js

```diff
@@ -148,101 +169,112 @@ class Stack {
       contents: {
         data: files,
         meta: {}
       },
-      parent: !!parent && normalizeFileFromPouchDB(parent)
+      parent: !!parent && this.normalizeFileFromPouchDB(parent)
     }
   }
 
-  getFolderContents = async (folderId, skip = 0, limit = FILES_FETCH_LIMIT) => {
-    const db = cozy.client.offline.getDatabase('io.cozy.files')
-    const resp = await db.find({
-      selector: {
-        dir_id: folderId,
-        name: { $gte: null },
-        type: { $gte: null }
-      },
-      use_index: this.folderIndex,
-      sort: ['dir_id', { type: 'desc' }, { name: 'desc' }],
-      limit,
-      skip
-    })
-    const files = resp.docs
-      .filter(f => f._id !== TRASH_DIR_ID)
-      .map(f => normalizeFileFromPouchDB(f))
-    return files
-  }
+  getFolderContents = (folderId, skip = 0, limit = FILES_FETCH_LIMIT) =>
+    this.getSortedFolder(folderId, 'name', 'asc', skip, limit)
 
   getSortedFolder = async (
     folderId,
     sortAttribute,
     sortOrder = 'asc',
     skip = 0,
-    limit = FILES_FETCH_LIMIT,
-    loadedFoldersCount = 0,
-    loadedFilesCount = 0
+    limit = FILES_FETCH_LIMIT
   ) => {
-    const shouldSeparateFilesAndFolders = sortAttribute === 'name'
-    const index = this.sortIndexes[sortAttribute]
-    if (!index) {
-      throw new Error(`Can't sort on ${sortAttribute}`)
-    }
-    const db = cozy.client.offline.getDatabase('io.cozy.files')
-    const query = (selector = {}, skipRows = skip) =>
-      db
-        .find({
-          selector: {
-            ...selector,
-            ...{
-              dir_id: folderId,
-              _id: { $ne: TRASH_DIR_ID },
-              [sortAttribute]: { $gte: null }
-            }
-          },
-          use_index: index,
-          sort: [{ [sortAttribute]: sortOrder }],
-          skip: skipRows,
-          limit
-        })
-        .then(resp => resp.docs.map(f => normalizeFileFromPouchDB(f)))
-
-    if (shouldSeparateFilesAndFolders) {
-      const folders = await query({
-        type: 'directory'
+    const isFirstLoad = skip === 0
+    let folders = []
+
+    if (isFirstLoad) {
+      const folderSortingOrder = sortAttribute === 'name' ? sortOrder : 'asc'
+
+      const allFolders = await this.query({
+        folderId,
+        type: 'directory',
+        sortAttribute: 'name',
+        sortOrder: folderSortingOrder,
+        skip: 0,
+        limit: null
       })
-      const files =
-        folders.length < limit
-          ? await query(
-              {
-                type: { $ne: 'directory' }
-              },
-              loadedFilesCount
-            )
-          : []
-      return [...folders, ...files]
-    } else {
-      return query()
+      folders = allFolders.filter(folder => folder._id !== TRASH_DIR_ID)
     }
+
+    const files = await this.query({
+      folderId,
+      type: 'file',
+      sortAttribute,
+      sortOrder,
+      skip,
+      limit
+    })
+    return [...folders, ...files]
```

Comment:
So same as above here — we load all the folders the first time, and then just the files.